ksp_plugin/interface_flight_plan.cpp

```diff
@@ -247,6 +247,35 @@ double principia__FlightPlanGetDesiredFinalTime(Plugin const* const plugin,
                  GetFlightPlan(*plugin, vessel_guid).desired_final_time()));
 }
 
+XYZ principia__FlightPlanGetGuidance(Plugin const* const plugin,
+                                     char const* const vessel_guid,
+                                     int const index) {
+  journal::Method<journal::FlightPlanGetGuidance> m(
+      {plugin, vessel_guid, index});
+  CHECK_NOTNULL(plugin);
+  auto const& manœuvre = GetFlightPlan(*plugin, vessel_guid).GetManœuvre(index);
+  Vector<double, Barycentric> barycentric_guidance;
+  if (manœuvre.is_inertially_fixed()) {
+    barycentric_guidance = manœuvre.InertialDirection();
+  } else {
+    auto const vessel_psychohistory_last =
+        plugin->GetVessel(vessel_guid)->psychohistory().last();
+    auto const to_manœuvre_frame =
+        manœuvre.frame()->ToThisFrameAtTime(vessel_psychohistory_last.time());
+    auto const from_manœuvre_frame =
+        to_manœuvre_frame.orthogonal_map().Inverse();
+    barycentric_guidance =
+        (from_manœuvre_frame *
+         manœuvre.frame()->FrenetFrame(
+             vessel_psychohistory_last.time(),
+             to_manœuvre_frame(
+                 vessel_psychohistory_last.degrees_of_freedom())).Forget())(
+            manœuvre.direction());
```

Comment:
Let's put this in Renderer and avoid complex code in the interface.  Look at FrenetToWorld for an example of a similar computation.